.github/workflows/rust.yml

```diff
@@ -6,11 +6,12 @@ on:
     branches: [ master, next ]
 jobs:
   build:
+    name: Run tests and doctests
     runs-on: ubuntu-latest
     strategy:
       matrix:
         rust: ["1.51.0", stable, beta, nightly]
```

Comment:
Feel free to bump MSRV here, adjust `Cargo.toml:package.rust-version` accordingly.

Reply:
My understanding is that, as this should be as low as possible for maximum compatibility, we should aim to never increase it. It is a minimum version. I suggest we only bump that version when someone wants to use a special new feature which is not supported in `1.51.0`. What do you think?

Reply:
We also have to increase our minimum supported version when our dependencies do. From the error posted above, it looks like nanorand now requires Rust 2021 which corresponds to Rust 1.56.0 or later

Reply:
Your understanding does agree with mine, but I've come to accept that _in practice_ we'll want to bump this version quicker and with our dependencies at least until `cargo` ships with an msrv-aware resolver algorithm. It also seems more brittle to claim compatibility with an untested Rust version, or to implement such resolver by-hand somehow. And, lastly, doing this is a useful testcase for the practability of our policy amyways. In particular if the risk of actual breakage is low, as it is here, because we do it by choice and without inherent need.

Reply:
Oh, I didn't see the `Cargo.toml` actually has a different version than the CI right now, of course now this makes sense!

Reply:
`cargo-msrv` says
```
Check for toolchain '1.57.0-x86_64-pc-windows-msvc' succeeded                                                                                                     2
Check for toolchain '1.56.1-x86_64-pc-windows-msvc' succeeded                                                                                                     7
   Finished The MSRV is: 1.56.1

```